sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -35,7 +50,12 @@ class JDBCOptions(
   val table = parameters("dbtable")
 
   // ------------------------------------------------------------
-  // Optional parameter list
+  // Optional parameters
+  // ------------------------------------------------------------
+  val driver = parameters.get(JDBCOptions.JDBC_DRIVER_CLASS)
```

Comment:
Sorry, I might not express it clearly. I mean move what we did for `driver` in `JdbcUtils.createConnectionFactory` to here.